Source/EventAnalysis/KTTMVAClassifier.cc

```diff
@@ -0,0 +1,149 @@
+/*
+ * KTTMVAClassifier.cc
+ *
+ *  Created on: Dec 13, 2016
+ *      Author: ezayas
+ */
+
+#include "KTTMVAClassifier.hh"
+
+#include "KTLogger.hh"
+
+#include "KTPowerFitData.hh"
+
+namespace Katydid
+{
+    KTLOGGER(evlog, "KTTMVAClassifier");
+
+    // Register the processor
+    KT_REGISTER_PROCESSOR(KTTMVAClassifier, "tmva-classifier");
+
+    KTTMVAClassifier::KTTMVAClassifier(const std::string& name) :
+            KTProcessor(name),
+            fMVAFile("foo.weights.xml"),
+            fAlgorithm("SomeAlgorithm"),
+            fMVACut(0.),
+            fClassifySignal("classify", this)
+    {
+        RegisterSlot( "power-fit", this, &KTTMVAClassifier::SlotFunctionPowerFitData );
+    }
+
+    KTTMVAClassifier::~KTTMVAClassifier()
+    {
+    }
+
+    bool KTTMVAClassifier::Configure(const scarab::param_node* node)
+    {
+        if (node == NULL) return false;
+
+        SetMVAFile(node->get_value< std::string >("mva-file", GetMVAFile()));
+        SetAlgorithm(node->get_value< std::string >("algorithm", GetAlgorithm()));
+        SetMVACut(node->get_value< double >("mva-cut", GetMVACut()));
+        
+        return true;
+    }
+
+    bool KTTMVAClassifier::ClassifyTrack( KTProcessedTrackData& trackData, double mva )
+    {
+        KTINFO(evlog, "Classifying track with MVA classifier = " << mva);
+        trackData.SetMVAClassifier( mva );
+
+        if( mva >= GetMVACut() )
+        {
+            KTDEBUG(evlog, "Classifying track as a signal peak");
+            trackData.SetMainband( true );
+        }
+        else
+        {
+            KTDEBUG(evlog, "Classifying track as a sideband peak");
+            trackData.SetMainband( false );
+        }
+
+        if( mva <= -999 )
+        {
+            KTWARN(evlog, "Classifier value is -999; something probably went wrong computing it");
+            return false;
+        }
+
+        return true;
+    }
+
+    void KTTMVAClassifier::SlotFunctionPowerFitData( Nymph::KTDataPtr data )
```

Comment:
Pretty much all of my comments from the dlib classifier apply here too:

* Why is this in a custom slot function?
* Why do you create a TMVA::Reader object every single time this is called?  It should be done once and kept in a member variable.
* Does the ClassifyTrack() function really need to exist as a separate function?